lib/mondrian/olap/connection.rb

```diff
@@ -378,6 +378,9 @@ def jdbc_uri_mssql
       }
 
       def jdbc_uri_sqlserver
+        if !@params[:port] && @params[:instance]
+          @params[:host] = "#{@params[:host]}\\#{@params[:instance]}"
```

Comment:
Do not modify `@params` as it is received as an argument in the `initialize` method and might create side effects.